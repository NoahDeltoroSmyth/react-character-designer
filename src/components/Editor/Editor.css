.head {
  margin: 5px;
}

.middle {
  margin: 5px;
}

.bottom {
  margin: 5px;
}

input {
  margin-top: 10px;
  margin-bottom: 10px;
}

select {
  width: 150px;
}

.catch-phrase {
  display: flex;
  flex-direction: column;
  width: 150px;
  margin: auto;
  margin-top: 10px;
  color: goldenrod;
}
