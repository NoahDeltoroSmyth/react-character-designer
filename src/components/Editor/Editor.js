import React from 'react';

import './Editor.css';

export default function Editor({
  head,
  setHead,
  middle,
  setMiddle,
  bottom,
  setBottom,
  phrase,
  setPhrase,
  setPhraseList,
}) {
  const handleClick = () => {
    setPhraseList((prevState) => [...prevState, phrase]);
    setPhrase('');
  };

  const handleSetHead = (e) => {
    setHead(e.target.value);
  };

  const handleSetMiddle = (e) => {
    setMiddle(e.target.value);
  };

  const handleSetBottom = (e) => {
    setBottom(e.target.value);
  };

  return (
    <div>
      <div className="head">
        <select value={head} onChange={(e) => setHead(e.target.value)}>
          <option value="bird-head">Bird</option>
          <option value="duck">Duck</option>
          <option value="dog">Dog</option>
          <option value="horse">Horse</option>
        </select>
      </div>
      <select value={middle} onChange={handleSetMiddle}>
        <option value="blue">Blue</option>
        <option value="fancy">Fancy</option>
        <option value="pink">Pink</option>
        <option value="red">Red</option>
      </select>
      <select value={bottom} onChange={(e) => setBottom(e.target.value)}>
        <option value="single-leg">Single Leg</option>
        <option value="white-pants">White Pants</option>
        <option value="blue-jeans">Blue Jeans</option>
      </select>
      <input value={phrase} onChange={handleSetBottom} type="text"></input>
      <button onClick={handleClick}>Add</button>
    </div>
  );
}
