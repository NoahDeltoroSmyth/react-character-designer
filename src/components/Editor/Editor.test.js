import { render } from '@testing-library/react';
import Editor from './Editor';

test('testing for classname editor', () => {
  const container = render(<Editor className="editor" />);
  expect(container).toMatchSnapshot();
});
