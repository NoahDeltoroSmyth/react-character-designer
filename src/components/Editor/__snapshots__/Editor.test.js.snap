// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing for classname editor 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="editor"
      >
        <div
          class="head"
        >
          <select>
            <option
              value="bird"
            >
              Bird
            </option>
            <option
              value="duck"
            >
              Duck
            </option>
            <option
              value="dog"
            >
              Dog
            </option>
            <option
              value="horse"
            >
              Horse
            </option>
          </select>
        </div>
        <div
          class="middle"
        >
          <select>
            <option
              value="blue"
            >
              Blue
            </option>
            <option
              value="dress"
            >
              Dress
            </option>
            <option
              value="pink"
            >
              Pink
            </option>
          </select>
        </div>
        <div
          class="bottom"
        >
          <select>
            <option
              value="leg"
            >
              Single Leg
            </option>
            <option
              value="white"
            >
              White Pants
            </option>
            <option
              value="blue"
            >
              Blue Pants
            </option>
          </select>
        </div>
        <div
          class="catch-phrase"
        >
          <label>
            Add a catchphrase!
          </label>
          <input
            type="text"
            value=""
          />
          <button>
            Add
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="editor"
    >
      <div
        class="head"
      >
        <select>
          <option
            value="bird"
          >
            Bird
          </option>
          <option
            value="duck"
          >
            Duck
          </option>
          <option
            value="dog"
          >
            Dog
          </option>
          <option
            value="horse"
          >
            Horse
          </option>
        </select>
      </div>
      <div
        class="middle"
      >
        <select>
          <option
            value="blue"
          >
            Blue
          </option>
          <option
            value="dress"
          >
            Dress
          </option>
          <option
            value="pink"
          >
            Pink
          </option>
        </select>
      </div>
      <div
        class="bottom"
      >
        <select>
          <option
            value="leg"
          >
            Single Leg
          </option>
          <option
            value="white"
          >
            White Pants
          </option>
          <option
            value="blue"
          >
            Blue Pants
          </option>
        </select>
      </div>
      <div
        class="catch-phrase"
      >
        <label>
          Add a catchphrase!
        </label>
        <input
          type="text"
          value=""
        />
        <button>
          Add
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
